docs/fee_design.md

```diff
@@ -0,0 +1,173 @@
+# Fee Exit Design
+
+This document describes the possible design for fee exits. It starts from the requirement and then describes the basic fee mechanism within our Plasma M(ore)VP design. It then proposes two different ways of having a fee transaction.
+
+
+## Requirement
+
+### Functional Requirement
+1. The operator can exit fees to an address the operator owns.
+2. Able to support fee exit for multiple transaction types where each transaction type can have different fee rules
+3. Support the initial rule for Payment transaction, which is a fixed amount in cents per transaction.
+4. Multiple rules can be set within a transaction type.
+   1. as a percentage of ETH gas. 
+   2. as a percentage of notional (aka transaction dollar amount).
+   3. fixed token price... floating with USD or other fiat.
+
+### Non-functional Requirement
+1. Fee exit does not take longer time than normal exit
+2. Fee exit can batch an exit for multiple fees collected in different transactions
+
+### Nice-to-have
+1. Able to A/B test fee mechanism
+
+### Out of Scope
+1. The fee rules are enforced by the operator by accepting transactions in the child chain and not by the smart contracts. A POA Plasma network always gives an operator the chance to censor transactions, therefore having the fee rule enforcement implemented in the smart contracts is not a design goal. However, users are protected by the watcher should the operator exit more fees then they have accepted in their transactions. 
+
+
+## General fee mechanism design
+
+### High level description
+Operator (Child chain) decides and enforces the fee format and the rule for transactions. The smart contracts do not check the rules for fee transactions. The rule enforcement is perfomed by the Child chain service. This is the case as long as the Plasma network runs in POA.
+
+The transaction fee is implied for a Payment transaction. It is the difference of the sum of inputs and the sum of outputs. For instance, if the sum of inputs is 10 ETH and sum of outputs is 9.9 Eth, the 0.1 Eth difference would be the transaction fee.
+
+In order exit fees, an operator would first put a special fee transaction into a plasma chain block. After that, the operator can spend that fee transaction output to a payment transaction. So the operator can exit it as a normal payment transaction. This removes the need to deploy an Exit Game contract for the fee transaction type as it requires only an empty contract to represent the existence of the transaction type and its protocol (using MoreVP instead of MVP).
+
+A fee transaction is special and different from a normal transaction in that:
+1. It does not need to consume any inputs. As the fee is implied (at least from the Payment transaction), there is no output that consumes an input.
+2. The verification of the fee transaction relies on child chain and watcher only. There are potentially different ways of doing it, which is described in a later section.
```

Comment:
There is only one way in this doc now as the non chosen design is removed. On the other hand, do we still have link/reference to it? Otherwise I think it might be better to leave it so people would know what we’ve been compared